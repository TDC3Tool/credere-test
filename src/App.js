import React from 'react'
import Slider from './components/Slider'
import sliderMock from './__mocks__/slider.json'

const App = () => {
  const { slides, configurations } = sliderMock

  return (
    <>
      <div className="container">
        <Slider
          duration={configurations.duration}
          effect={configurations.effect}
          autoPlay={configurations.autoPlay}
          slides={slides}
        />
      </div>
    </>
  )
}

export default App
