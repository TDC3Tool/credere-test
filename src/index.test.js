import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'

describe('App', () => {
  it('should render App', () => {
    const element = renderer.create(<App />).toJSON()
    expect(element).toMatchSnapshot()
  })
})
