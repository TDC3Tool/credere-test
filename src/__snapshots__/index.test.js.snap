// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `
Array [
  <div
    className="container"
  >
    <h2
      className="sc-htoDjs gsDwee"
    >
      Form
    </h2>
    <form
      className="sc-kpOJdX fLqRtU"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <h3
          className="sc-dnqmqq bJaIPW"
        >
          Cliente
        </h3>
        <div>
          <div
            className="grid user-details "
          >
            <div
              className="border-sizing sc-kAzzGY izUUKG"
            >
              <label
                className="sc-chPdSV boxXeJ"
                htmlFor="name"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                >
                  Nome
                </span>
                 *
              </label>
              <input
                className="sc-kgoBCf iSFPjC"
                id="name"
                name="name"
                onChange={[Function]}
                required={true}
              />
              <div
                className="sc-kGXeez dNDVLu"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                />
              </div>
            </div>
            <div
              className="border-sizing sc-kAzzGY izUUKG"
            >
              <label
                className="sc-chPdSV boxXeJ"
                htmlFor="birthday"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                >
                  Data de nascimento
                </span>
                 *
              </label>
              <input
                className="sc-kgoBCf iSFPjC"
                id="birthday"
                max="2019-07-18"
                name="birthday"
                onChange={[Function]}
                required={true}
                type="date"
              />
              <div
                className="sc-kGXeez dNDVLu"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                />
              </div>
            </div>
          </div>
          <div
            className="sc-dxgOiQ TFlRf"
          >
            <span
              className="sc-gZMcBi giNnHm"
            >
              Carteira de motorista
            </span>
            <div
              className="grid drive-license-fields"
            >
              <div
                className="border-sizing sc-kAzzGY izUUKG"
              >
                <input
                  className="sc-kgoBCf iSFPjC"
                  id="driver_license"
                  name="driver_license"
                  onChange={[Function]}
                  pattern="^[0-9]*"
                  required={true}
                />
                <div
                  className="sc-kGXeez dNDVLu"
                >
                  <span
                    className="sc-gZMcBi giNnHm"
                  />
                </div>
              </div>
              <div
                className="border-sizing sc-kAzzGY izUUKG"
              >
                <input
                  className="sc-kgoBCf iSFPjC"
                  id="driver_license_issue"
                  max="2019-07-18"
                  name="driver_license_issue"
                  onChange={[Function]}
                  required={true}
                  type="date"
                />
                <div
                  className="sc-kGXeez dNDVLu"
                >
                  <span
                    className="sc-gZMcBi giNnHm"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="grid user-address"
          >
            <div
              className="border-sizing sc-kAzzGY izUUKG"
            >
              <label
                className="sc-chPdSV boxXeJ"
                htmlFor="state"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                >
                  Estado
                </span>
                 *
              </label>
              <input
                className="sc-kgoBCf iSFPjC"
                id="state"
                maxLength="2"
                name="state"
                onChange={[Function]}
                pattern="[A-Za-z]{2}"
                required={true}
              />
              <div
                className="sc-kGXeez dNDVLu"
              >
                <span
                  className="sc-gZMcBi giNnHm"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-ckVGcZ foYhJp"
      >
        <h3
          className="sc-dnqmqq bJaIPW"
        >
          Telefones
        </h3>
      </div>
      <div
        className="sc-jKJlTe jxfROG"
      >
        <h3
          className="emails-title sc-dnqmqq bJaIPW"
        >
          E-mails
        </h3>
        <div
          className="grid emails-list"
        >
          <span
            className="sc-gZMcBi giNnHm"
          >
            Não há e-mails cadastrados!
          </span>
        </div>
        <div
          className="grid emails-form"
        >
          <button
            className="button-md button-outline-success transition-all sc-cSHVUG ljIwnR"
            onClick={[Function]}
            type="button"
          >
            Adicionar mais
          </button>
        </div>
      </div>
      <button
        className="button-lg button-normal-default transition-all sc-cSHVUG ljIwnR"
        onClick={[Function]}
        type="submit"
      >
        Register
      </button>
    </form>
  </div>,
  <div
    className="container"
  >
    <h2
      className="sc-htoDjs gsDwee"
    >
      Slider Default
    </h2>
    <div
      className="relative sc-bdVaJa bJmnKC"
    >
      <div
        className="full-content relative sc-bwzfXH uTxCW"
      >
        <div
          className="visible transition-all absolute sc-htpNat dxOMxF"
          data-index={0}
          data-template="default"
        >
          <div
            className="grid full-content sc-gqjmRU kIqdMS"
          >
            <div
              className="flex-column-center relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10153244993210.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. Nulla et nisl eget nulla imperdiet mattis?
                </h3>
              </div>
            </div>
            <div
              className="flex-column-center sc-gzVnrw dkdosU"
            >
              <p
                className="sc-iwsKbI ibDZNz"
                style={
                  Object {
                    "color": "#333333",
                  }
                }
              >
                Aliquam facilisis tempus neque, ac accumsan elit bibendum non. Duis molestie libero porta tortor placerat, vitae cursus diam placerat. Fusce feugiat libero erat, suscipit venenatis purus eleifend non
              </p>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={1}
          data-template="template_1"
        >
          <div
            className="full-content grid sc-jTzLTM hfhqii"
          >
            <div
              className="flex-algin-end full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10091423089001.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
            </div>
            <div>
              <div
                className="sc-fjdhpX dJvaXu"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#333333",
                    }
                  }
                >
                  Lorem ipsum dolor
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#999999",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. 
                </h3>
              </div>
              <div
                className="flex-column-center sc-gzVnrw dkdosU"
              >
                <p
                  className="sc-iwsKbI ibDZNz"
                  style={
                    Object {
                      "color": "#333333",
                    }
                  }
                >
                  Aliquam facilisis tempus neque, ac accumsan elit bibendum non. Duis molestie libero porta tortor placerat, vitae cursus diam placerat. Fusce feugiat libero erat, suscipit venenatis purus eleifend non
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={2}
          data-template="template_2"
        >
          <div
            className="full-content sc-jzJRlG bhbMmg"
          >
            <div
              className="flex-algin-end full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10145408149192.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. Nulla et nisl eget nulla imperdiet mattis?
                </h3>
              </div>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={3}
          data-template="template_3"
        >
          <div
            className="full-content sc-VigVT gNgpkb"
          >
            <div
              className="flex-column-center full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/06/17/17152604467364.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="absolute sc-bxivhb dEKikf"
        data-max={0}
      >
        <button
          className="button-lg button-normal-default transition-all sc-cSHVUG ljIwnR"
          onClick={[Function]}
          type="button"
        >
          Próximo
        </button>
      </div>
    </div>
  </div>,
  <div
    className="container"
  >
    <h2
      className="sc-htoDjs gsDwee"
    >
      Slider Auto Play (5s)
    </h2>
    <div
      className="relative sc-bdVaJa bJmnKC"
    >
      <div
        className="full-content relative sc-bwzfXH uTxCW"
      >
        <div
          className="visible transition-all absolute sc-htpNat dxOMxF"
          data-index={0}
          data-template="default"
        >
          <div
            className="grid full-content sc-gqjmRU kIqdMS"
          >
            <div
              className="flex-column-center relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10153244993210.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. Nulla et nisl eget nulla imperdiet mattis?
                </h3>
              </div>
            </div>
            <div
              className="flex-column-center sc-gzVnrw dkdosU"
            >
              <p
                className="sc-iwsKbI ibDZNz"
                style={
                  Object {
                    "color": "#333333",
                  }
                }
              >
                Aliquam facilisis tempus neque, ac accumsan elit bibendum non. Duis molestie libero porta tortor placerat, vitae cursus diam placerat. Fusce feugiat libero erat, suscipit venenatis purus eleifend non
              </p>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={1}
          data-template="template_1"
        >
          <div
            className="full-content grid sc-jTzLTM hfhqii"
          >
            <div
              className="flex-algin-end full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10091423089001.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
            </div>
            <div>
              <div
                className="sc-fjdhpX dJvaXu"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#333333",
                    }
                  }
                >
                  Lorem ipsum dolor
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#999999",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. 
                </h3>
              </div>
              <div
                className="flex-column-center sc-gzVnrw dkdosU"
              >
                <p
                  className="sc-iwsKbI ibDZNz"
                  style={
                    Object {
                      "color": "#333333",
                    }
                  }
                >
                  Aliquam facilisis tempus neque, ac accumsan elit bibendum non. Duis molestie libero porta tortor placerat, vitae cursus diam placerat. Fusce feugiat libero erat, suscipit venenatis purus eleifend non
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={2}
          data-template="template_2"
        >
          <div
            className="full-content sc-jzJRlG bhbMmg"
          >
            <div
              className="flex-algin-end full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/07/10/10145408149192.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna. Nulla et nisl eget nulla imperdiet mattis?
                </h3>
              </div>
            </div>
          </div>
        </div>
        <div
          className="hidden transition-all absolute sc-htpNat dxOMxF"
          data-index={3}
          data-template="template_3"
        >
          <div
            className="full-content sc-VigVT gNgpkb"
          >
            <div
              className="flex-column-center full-content relative"
              style={
                Object {
                  "backgroundImage": "url(https://img.ibxk.com.br/2019/06/17/17152604467364.jpg)",
                }
              }
            >
              <div
                className="full-content absolute sc-EHOje fZgOQP"
              />
              <div
                className="relative sc-bZQynM sMRAj"
              >
                <h2
                  className="sc-htoDjs kexmgL"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Lorem ipsum dolor sit amet
                </h2>
                <h3
                  className="sc-dnqmqq bJaIPW"
                  style={
                    Object {
                      "color": "#FFFFFF",
                    }
                  }
                >
                  Sed non dui sed purus hendrerit aliquet a nec magna
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="absolute sc-bxivhb dEKikf"
        data-max={0}
      >
        <button
          className="button-lg button-normal-default transition-all sc-cSHVUG ljIwnR"
          onClick={[Function]}
          type="button"
        >
          Próximo
        </button>
      </div>
    </div>
  </div>,
]
`;
